feat: add embed builder and migrate current commands to embeds

Niggelgame opened the pull request:
resolves #38

Niggelgame committed:
feat: add embed builder and migrate current commands to embeds

DRSchlaubi commented:
Run `gradlew ktlintFormat` to fix the CI

DRSchlaubi commented on src/main/kotlin/dev/schlaubi/votebot/commands/ClaimPermissionsCommand.kt:
```diff
@@ -35,7 +36,7 @@ object ClaimPermissionsCommand : SingleCommand() {
     override suspend fun execute(context: Context) {
         if (Permission.Administrator !in context.executor.asMember().getPermissions()) {
-            context.respond("You need the `ADMINISTRATOR` permission to execute this command")
+            context.respond(Embeds.error("You need the `ADMINISTRATOR` permission to execute this command", description = null))
```
The description is null per default

Niggelgame replied:
yeah, forgot to remove it again since at the beginning it was not null by default

DRSchlaubi commented on src/main/kotlin/dev/schlaubi/votebot/core/VoteBotImpl.kt:
```diff
@@ -19,6 +19,7 @@
 
 package dev.schlaubi.votebot.core
 
+import dev.kord.common.entity.PresenceStatus
```
Why was this added?

DRSchlaubi commented on src/main/kotlin/dev/schlaubi/votebot/util/Colors.kt:
```diff
@@ -0,0 +1,43 @@
+@Suppress("KDocMissingDocumentation", "unused", "MagicNumber")
+object Colors {
+    // Discord
+    val BLURLPLE: Color = Color(114, 137, 218)
```
Those are actually wrong update them to match the new branding

Niggelgame replied:
Should I remove all the non-existing colors?

Niggelgame replied:
Or e.g. replace the current LIGHT_GREEN with the branding GREEN

Niggelgame replied:
@DRSchlaubi

DRSchlaubi commented on src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt:
```diff
@@ -0,0 +1,213 @@
+     * @see EmbedBuilder
+     */
+    fun info(title: String, description: String? = null, builder: EmbedCreator = {}): EmbedBuilder =
+        EmbedBuilder().apply {
```
use the global scope embed function here instead

DRSchlaubi commented on src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt:
```diff
@@ -0,0 +1,213 @@
+            if (children.isNotEmpty()) {
+                field {
+                    name = "Aliase"
+                    value = children.joinToString("`, `", "`", "`")
+                }
+            }
+            field {
+                name = "Usage"
+                value = formatCommandUsage(command)
+            }
+            field {
+                name = "Permission"
+                value = command.permission.toString()
+            }
+//            addField("Permission", command.permission.name)
+//            val subCommands = command.registeredCommands.map(::formatSubCommandUsage)
+//            if (subCommands.isNotEmpty()) {
+//                addField("Sub commands", subCommands.joinToString("\n"))
+//            }
+        }
+    }*/
+
+    /*private fun formatCommandUsage(command: Command<*>): String {
+        val usage = command.arguments.joinToString(" ") {
+            if ("optional" in it.toString()) "[${it.name}]" else "<${it.name}>"
+        }
+
+        return "!${command.name} $usage"
+    }*/
+//
+//    private fun formatSubCommandUsage(command: AbstractSubCommand): String {
+//        val builder = StringBuilder(Constants.firstPrefix)
+//        builder.append(' ').append(command.name).append(' ').append(command.usage.replace("\n", "\\n"))
+//
+//        val prefix = " ${command.parent.name} "
+//        builder.insert(Constants.firstPrefix.length, prefix)
+//        builder.append(" - ").append(command.description)
+//
+//        return builder.toString()
+//    }
```
remove this as help command isn't needed for slash commansd

DRSchlaubi commented on src/main/kotlin/dev/schlaubi/votebot/util/Embeds.kt:
```diff
@@ -0,0 +1,213 @@
+    }
+
+
+/*
+    */
+    /**
+     * Responds to this command with an embed provided by [base] and applies [creator] to it
+     *//*
+    @Deprecated("Use sendResponse instead", ReplaceWith("sendResponse(base, creator)"))
+    suspend fun KordEvent.respondEmbed(
+        base: EmbedBuilder,
+        creator: suspend EmbedBuilder.() -> Unit = {}
+    ): Message {
+        return respond {
+            creator(base)
+            embed = base
+        }
+    }*/
+}
```
this can be removed too

Niggelgame committed:
fix: change code style to pass style test

Niggelgame committed:
fix: remove comments and use global embed function

DRSchlaubi commented:
> The base branch requires all commits to be signed. Learn more about signing commits.

DUDE

Niggelgame committed:
feat: add new branding colors

Niggelgame commented:
freak i used the wrong git profile

DRSchlaubi commented:
> freak i used the wrong git profile

just rebase
squash
and sign that

Niggelgame committed:
feat: add embed builder and migrate current commands to embeds

fix: change code style to pass style test

fix: remove comments and use global embed function

feat: add new branding colors

Niggelgame commented:
actually forgot the squash

but fr took long enough for me to rebase on my own branch... until i found the IJ command to rebase on the root of the branch
